### ISSUE TITLE
aws_ssm_parameter fails to create parameters in AWS China, provider version 2.10

### ISSUE THREAD

**Issue by smolse** [terraform-providers/terraform-provider-aws#8662]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.11.13
+ provider.aws v2.10.0
```

### Affected Resource(s)

* aws_ssm_parameter

### Terraform Configuration Files

```hcl
provider "aws" {
  version = "~> 2.10"

  profile = "<profile_name>"
  region = "cn-north-1"
}

resource "aws_ssm_parameter" "test_param" {
  name  = "test"
  value = "test"
  type  = "String"
}
```

### Expected Behavior

Terraform should put a parameter to the Parameter Store.

### Actual Behavior

Terraform fails to put a parameter to the Parameter Store.

### Steps to Reproduce

```
$ terraform init
```
```
$ terraform plan
```
```
$ terraform apply

An execution plan has been generated and is shown below.
provider "aws" {
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + aws_ssm_parameter.test_param
      id:     <computed>
      arn:    <computed>
      key_id: <computed>
      name:   "test"
      tier:   "Standard"
      type:   "String"
      value:  <sensitive>


Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_ssm_parameter.test_param: Creating...
  arn:    "" => "<computed>"
  key_id: "" => "<computed>"
  name:   "" => "test"
  tier:   "" => "Standard"
  type:   "" => "String"
  value:  "<sensitive>" => "<sensitive>"

Error: Error applying plan:

1 error(s) occurred:

* aws_ssm_parameter.test_param: 1 error(s) occurred:

* aws_ssm_parameter.test_param: error creating SSM parameter: ValidationException: Tier is not supported in region cn-north-1.
	status code: 400, request id: a8b9ed27-b083-4241-ab0d-caed3cc0587b
```

### References

The error is presumably caused by the recent changes from #8525.

There are some differences in AWS China and one of them is that there is no standard tier for Parameter Store. As the result `AmazonSSM.PutParameter` API target does not accept `Tier` argument at all and throws an error whenever it is present in the request payload.

**Comment by bflad** [comment]
Hi @smolse 👋 Sorry for this unexpected behavior! The maintainers at HashiCorp currently only have access to AWS Commercial and AWS GovCloud (US) at the moment so API differences in AWS China are hard to determine when adding new functionality that works in the other two AWS partitions. 😖 (If you have a good relationship with AWS China, they can reach out to HashiCorp through the AWS Partner Network to potentially work out a solution with us this problem.)

We'll likely need to update the resource code in this case to try the API call with the `Tier` parameter as it does today, but retry the call without it if this specific error occurs. This will allow the feature to automatically start working in AWS China without code chances when support is added.

Until this code change is merged and released, the workaround within Terraform would be use a Terraform AWS Provider version that precedes when this functionality was added in [version 2.10.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2100-may-10-2019), e.g.

```hcl
provider "aws" {
  # ... potentially other configuration ...
  version = "2.9.0"
}
```

Hope this helps and apologize for the hassle.

**Comment by davidcelis** [comment]
@bflad I think a good solution could be to remove the default of `"Standard"` and only submit the tier when it's provided in the TF resource… From what I can tell, the AWS API itself sets `"Standard"` as the default for tier when tiers are supported. I included the default in the provider for consistency but it's not really necessary

**Comment by bflad** [comment]
@davidcelis for situations like these where AWS partitions differ in abilities by which version of the AWS service is running, we typically prefer to "think ahead" with our Terraform resource code to do the correct behavior with the most up-to-date partition (almost always AWS Commercial) to simplify maintenance in the future.

In this case, we can default to proactively setting the Terraform state to `Standard` for the `tier` attribute if its omitted from the API response, which saves us from needing to setup any `DiffSuppressFunc`, using `Computed: true` which disables automatic drift detection, or partition-specific logic that will eventually become outdated in the future. In the context of Sentinel policies, keeping the default of `Standard` in the state also means that only one policy condition (e.g. not worrying about "") needs to be written for use across all AWS partitions.

**Comment by bflad** [comment]
The workaround for AWS China has been merged and will release with version 2.11.0 of the Terraform AWS Provider, very shortly. 👍

**Comment by bflad** [comment]
This has been released in [version 2.11.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2110-may-17-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
